### PR TITLE
fix: reset accept fut in stop

### PR DESCRIPTION
**Pull request by Ivansete-status**
Changes:

- Reset `acceptFuts` seq when `stop` is invoked: this is needed to properly perform connection restarts between nodes and this issue was detected in `nwaku` tests, as explained in https://github.com/waku-org/nwaku/issues/2145

- Add a debug log when an exception is caught. This change is not directly needed by the "reconnect" issue mentioned in the previous point.

### CODE REVIEW

File: libp2p/transports/tcptransport.nim [hunk 1]
```diff
@@ -221,6 +222,7 @@ method accept*(self: TcpTransport): Future[Connection] {.async.} =
       let observedAddr = MultiAddress.init(transp.remoteAddress).tryGet()
       return await self.connHandler(transp, Opt.some(observedAddr), Direction.In)
     except CancelledError as exc:
+      debug "CancelledError", exc = exc.msg
```
Comment (diegomrsantos) [inline libp2p/transports/tcptransport.nim:225]: I believe this is logged by the caller as it is re-raised.

File: libp2p/transports/tcptransport.nim [hunk 2]
```diff
@@ -190,6 +190,7 @@ method stop*(self: TcpTransport) {.async.} =
     await allFutures(toWait)
 
     self.servers = @[]
+    self.acceptFuts = @[]
```
Comment (diegomrsantos) [inline libp2p/transports/tcptransport.nim:193]: I don't know if this is the right approach as I don't have much knowledge about this code. @dryajov could you please advise?

Comment (dryajov) [reply libp2p/transports/tcptransport.nim:193]: We need to make sure that all the accept futs are properly canceled, before resetting.

File: libp2p/transports/tcptransport.nim [hunk 3]
```diff
@@ -191,6 +191,9 @@ method stop*(self: TcpTransport) {.async.} =
 
     self.servers = @[]
 
+    if self.acceptFuts.allIt(it.finished()):
```
Comment (diegomrsantos) [inline libp2p/transports/tcptransport.nim:194]: I interpreted @dryajov's comment as: for each future that hasn't been finished, it must be canceled.

**Review by diegomrsantos — DISMISSED** [review]
LGTM, many thanks.

File: libp2p/transports/tcptransport.nim (hunk 3 above)
Comment (dryajov) [reply libp2p/transports/tcptransport.nim:194]: Yes, that should be enough. So this new check isn't necessary, the previous version without this change should be enough.

Comment (Ivansete-status) [reply libp2p/transports/tcptransport.nim]: > Yes, that should be enough. So this new check isn't necessary, the previous version without this change should be enough.

Okay perfect! I just reverted the previous commit.

@diegomrsantos - Kindly re-approve and merge if you consider it appropriate :)
Thanks!

Comment (diegomrsantos) [reply libp2p/transports/tcptransport.nim]: Sure, sorry for the trouble.

**Review by diegomrsantos — APPROVED** [review]

File: libp2p/transports/tcptransport.nim (hunk 3 above)
Comment (Ivansete-status) [reply libp2p/transports/tcptransport.nim]: > Sure, sorry for the trouble.

haha no worries at all! Thanks for the patience! :raised_hands:

**Pull request merged**